plugins/modules/edpm_nova_derive_pci_passthrough_whitelist.py

```diff
@@ -0,0 +1,671 @@
+#!/usr/bin/python3
+
+#
+# Copyright 2024 Red Hat Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+__metaclass__ = type
+
+import json
+import os
+import re
+import yaml
+
+
+from ansible.module_utils.basic import AnsibleModule
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.0',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: edpm_nova_derive_pci_passthrough_whitelist
+author:
+    - OpenStack EDPM Contributors
+version_added: '1.0'
+short_description: Derive pci passthrough whitelist for nova from sriov device spec
+notes: []
+description:
+  - This module updates the pci whitelist addr list for nova,
+    when sriov nic partitioning is configured in EDPM nodes.
+    See openstack user guide for details of
+    NovaPCIPassthrough format.
+options:
+  sriov_nova_conf_file:
+    description:
+      - User specified nova conf file for nic device lookup
+    type: str
+    required: true
+    default: '/var/lib/openstack/config/nova/03-sriov-nova.conf'
+  neutron_sriov_conf_file:
+    description:
+      - User specified nova conf file for nic device lookup
+    type: str
+    required: false
+    default: '/var/lib/config-data/ansible-generated/neutron-sriov-agent/01-neutron-sriov-agent.conf'
+  save_conf_file:
+    description:
+      - If enabled, the derived file will be saved (backward compatible)
+    type: bool
+    required: false
+    default: false
+'''
+
+EXAMPLES = '''
+- name: Create pci whitelist from defaults
+  edpm_nova_derive_pci_passthrough_whitelist:
+    - 03-sriov-nova.conf
+'''
+
+RETURN = '''
+'''
+
+_PASSTHROUGH_WHITELIST_KEY = 'device_spec'
+_PCI_DEVICES_PATH = '/sys/bus/pci/devices'
+_SYS_CLASS_NET_PATH = '/sys/class/net'
+_DERIVED_PCI_WHITELIST_FILE = '/var/lib/openstack/pci_passthrough_whitelist.json'
```

Comment:
again this si not an accpaable name but also I'm not sure this is a good location to use.

if this file was owned by the nova roles then it should be in /var/lib/openstack/config/nova

Reply:
This was just for backward compatible, where the derived pci addr would be written if "save_conf_file" is set. The actual update will be done in "03-sriov-nova.conf"